pkg/machinery/overlay/adapter/adapter.go

```diff
@@ -0,0 +1,70 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package adapter provides an adapter for the overlay installer.
+package adapter
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/siderolabs/talos/pkg/machinery/overlay"
+)
+
+// Execute executes the overlay installer.
+func Execute(installer overlay.Installer) {
+	if len(os.Args) < 2 {
```

Comment:
we could tidy up things a bit by calling from this function a function which returns an error:

```go
if err := execute(installer); err != nil {
   fmt.Fprint(os.Stderr, err.Error())
   os.Exit(1)
}
// success
```

this way we don't need to repeat this everywhere, and just return an error